src/redux/watchList/reducer.js

```diff
@@ -54,7 +54,18 @@ export default (state = initalState, action) => {
       return {
         ...state,
         counter: state.counter + 1,
-        watchlists: [...state.watchlists, { ...action.payLoad, id: state.counter }],
+        watchlists: [
+          ...state.watchlists,
+          {
+            id: state.counter + 1,
+            name: action.payLoad,
+            stocks: [],
+          }],
+      };
+    case DELETE_WATCHLIST:
+      return {
+        counter: state.counter - 1,
+        watchlists: [...state.watchlists.filter((item) => item.id !== action.payLoad)],
```

Comment:
you dont need to destruct the original state, you can filter it directly, filter doesnt change the original array